Assets/HoloToolkit/SpatialMapping/Scripts/SpatialMappingObserver.cs

```diff
@@ -107,22 +78,47 @@ protected override void Awake()
         /// </summary>
         private void Update()
         {
-            // Only do processing if the observer is running.
-            if (ObserverState == ObserverStates.Running)
+            if ((ObserverState == ObserverStates.Running) && (outstandingMeshRequest == null))
             {
-                // If we don't have mesh creation in flight, but we could schedule mesh creation, do so.
-                if (surfaceWorkOutstanding == false && surfaceWorkQueue.Count > 0)
+                if (surfaceWorkQueue.Count > 0)
                 {
-                    // Pop the SurfaceData off the queue.  A more sophisticated algorithm could prioritize
+                    // We're using a simple first-in-first-out rule for requesting meshes, but a more sophisticated algorithm could prioritize
                     // the queue based on distance to the user or some other metric.
-                    SurfaceData surfaceData = surfaceWorkQueue.Dequeue();
+                    var surfaceID = surfaceWorkQueue.Dequeue();
+
+                    // TODO: consider pooling objects for reuse
```

Comment:
I couldn't measure a perf difference, but I went ahead and added it back in out of paranoia.